STM32/AC/Core/Src/main.c

```diff
@@ -257,34 +251,6 @@ static void MX_ADC1_Init(void)
 
 }
 
-/**
-  * @brief IWDG Initialization Function
-  * @param None
-  * @retval None
-  */
-static void MX_IWDG_Init(void)
-{
-
-  /* USER CODE BEGIN IWDG_Init 0 */
-
-  /* USER CODE END IWDG_Init 0 */
-
-  /* USER CODE BEGIN IWDG_Init 1 */
-
-  /* USER CODE END IWDG_Init 1 */
-  hiwdg.Instance = IWDG;
-  hiwdg.Init.Prescaler = IWDG_PRESCALER_4;

```

Comment:
Did you try changing the IWDG prescalar value before removing it? The `IWDG_PRESCALER_4` is the strictest option. 

I am almost convinced that choosing a higher prescalar would solve the IWDG triggering during flashing.